File: CineBox/CineBox.Model/ViewModels/Hall.cs
using System;
namespace CineBox.Model.ViewModels
{
	public partial class Hall
	{
        public int Id { get; set; }

        public int CinemaId { get; set; }

        public string Name { get; set; } = null!;

        public virtual Cinema Cinema { get; set; } = null!;
    }
}

File: CineBox/UI/cinebox_mobile/lib/models/Hall/hall.dart
import 'package:cinebox_mobile/models/Cinema/cinema.dart';
import 'package:json_annotation/json_annotation.dart';

part 'hall.g.dart';

@JsonSerializable()
class Hall {
  int? id;
  int? cinemaId;
  String? name;
  Cinema? cinema;

  Hall();

  factory Hall.fromJson(Map<String, dynamic> json) => _$HallFromJson(json);

  Map<String, dynamic> toJson() => _$HallToJson(this);
}
